### DISCUSSION TITLE
Dependency apt installs

### DISCUSSION THREAD

**Discussion by foxhoundv** [nextcloud/all-in-one#2003] [Ideas] [answered]
Hello, I was trying to install the app "PDF Downloader", after install there was some notifications...

![image](https://user-images.githubusercontent.com/16512250/219430666-263744b4-3cf1-4fbe-a4bf-5cb027458984.png)

I know that when a container gets updated after you have gone in to it and ran "apt install *aptnamehere*", you will loose it and have to reinstall it. Is there a way that if certain dependencies are needed, that they can be added to a special container or persistent volume for these purposes? I know that the installation of apts can go in various areas but was wondering if there was a way to catch that so that if a command was given it can follow paths and merge them upon container updates.

I don't know if there is an easier way to go about this unless it gets applied to the image. Which is horrible for image creators because then the image gets bigger and bigger to accommodate everyone's needs. Yet if there was a way to keep the image small but still allow for special situations like this... I don't know, just an idea.

**Comment by Zoey2936** [comment, marked as answer]
See: https://github.com/nextcloud/all-in-one#how-to-add-os-packages-permanently-to-the-nextcloud-container

**Reply by foxhoundv** [reply]
Thank you for the reply, I looked at that and followed the link to the list. The one item that PDF Downloader requires isn't found in the list provided(specifically pdftk). Does that mean it won't work?

**Reply by Zoey2936** [reply]
alpine remove it two years ago: https://gitlab.alpinelinux.org/alpine/aports/-/commits/master?search=pdftk

**Reply by foxhoundv** [reply]
Bummer... Thanks!
